### DISCUSSION TITLE
Cuda 11.7 Pytorch Compatibility

### DISCUSSION THREAD

**Discussion by Shifter2600** [mindee/doctr#1341] [Q&A] [answered]
Hi Community.
I'm using the base nvidia/cuda:11.7.1-devel-ubuntu22.04 Image and installing Python and the doctr `torch` version.
I'm getting this error which is leading me to Pytorch version requirements. Is there a way to get doctr working on a compatible version that uses Cuda 11.7?

/opt/python/lib/python3.10/site-packages/torch/cuda/__init__.py:138: UserWarning: CUDA initialization: The NVIDIA driver on your system is too old (found version 11070). Please update your GPU driver by downloading and installing a new version from the URL: http://www.nvidia.com/Download/index.aspx Alternatively, go to: https://pytorch.org to install a PyTorch version that has been compiled with your version of the CUDA driver. (Triggered internally at ../c10/cuda/CUDAFunctions.cpp:108.)
  return torch._C._cuda_getDeviceCount() > 0

**Comment by felixdittrich92** [comment]
Hi @Shifter2600 :wave:,

Your posted error has nothing to do with doctr it's raissed by pytorch. doctr works well with cuda 11.7 :) 
You should  update your host machine gpu drivers and check the installed cuda version.


BTW: We provide prebuild images at: https://github.com/mindee/doctr/pkgs/container/doctr

**Reply by Shifter2600** [reply]
I will try using ghcr.io/mindee/doctr:torch-py3.9.18-gpu-2023-09 as my base OS and I believe everything is up to date for my GPU.

+-----------------------------------------------------------------------------+
| NVIDIA-SMI 515.43.04    Driver Version: 515.43.04    CUDA Version: 11.7     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  Off  | 00000000:0A:00.0 Off |                  N/A |
| 24%   56C    P2    65W / 275W |   6747MiB / 11264MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

**Reply by Shifter2600** [reply, marked as answer]
Just reporting back using ghcr.io/mindee/doctr:torch-py3.9.18-gpu-2023-09 as my base OS worked.
